proxy/grpcproxy/kv.go

```diff
@@ -35,8 +35,8 @@ func NewKvProxy(c *clientv3.Client) pb.KVServer {
 }
 
 func (p *kvProxy) Range(ctx context.Context, r *pb.RangeRequest) (*pb.RangeResponse, error) {
-	// if request set Serializable, serve it from local cache first
-	if r.Serializable {
+	// if request set serializable or a revision is specified, serve it from local cache first
+	if r.Serializable || r.Revision != 0 {
```

Comment:
`r.Revision != 0 && !r.Serializable` needs to go through consensus if it's not already in the cache. Giving a rev doesn't guarantee it's been committed to all members yet.